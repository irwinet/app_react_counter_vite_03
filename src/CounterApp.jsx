/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
import PropTypes from 'prop-types';
import { useState } from 'react';

export const CounterApp = ({ value }) => {
  
  const [counter, setCounter] = useState(value);

  // eslint-disable-next-line no-unused-vars
  const handleAdd = (event) => {
    // console.log(event)
    setCounter(counter+1);
    // setCounter((c) => c+1);
  }  

  const handleSubtract = () => setCounter(counter-1);

  const handleReset= () => setCounter(value);

  return (
    <>
        <h1>CounterApp</h1>
        <h2> { counter } </h2>
        <button onClick={ handleAdd } className='btn btn-primary'>
            +1
        </button>
        <button onClick={ handleSubtract } className='btn btn-primary'>
            -1
        </button>
        <button aria-label="btn-reset" onClick={ handleReset } className='btn btn-primary'>
            Reset
        </button>
    </>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,    
}
