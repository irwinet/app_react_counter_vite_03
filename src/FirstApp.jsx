/* eslint-disable react/prop-types */
// import { Fragment } from 'react'

// export const FirstApp = () => {
//   return (
//     <Fragment>
//         <h1>Irwin</h1>
//         <p>Soy un subtitulo</p>
//     </Fragment>
//   )
// }

// const newMessage = { key: 'hola' };

import PropTypes from 'prop-types';

const getSaludo = (name) => 'Hola: ' + name;

export const FirstApp = ({title, subTitle, name}) => {
    // console.log(props);

    return (
        <>
            {/* <code>{ JSON.stringify(newMessage) }</code> */}
            <h1 data-testid="test-title">{ title }</h1>
            <h1>{getSaludo('Irwin')}</h1>
            <p>{subTitle}</p>
            <p>{subTitle}</p>
            <p>{name}</p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,    
    subTitle: PropTypes.number.isRequired,    
}

FirstApp.defaultProps = {
    name: 'Irwin',
    subTitle: 1,
    title: 'Titulo Default',
}