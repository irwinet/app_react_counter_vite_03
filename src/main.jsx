
import React from 'react'
import ReactDOM from 'react-dom/client'
// import { HelloWorldApp } from './HelloWorldApp'
// import { FirstApp } from './FirstApp'
// import { App } from './HelloWorldApp'
// import App from './HelloWorldApp'
import { CounterApp } from './CounterApp'
import './styles.css'

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <FirstApp title='Titulo' subTitle={123} /> */}
        <CounterApp value={10} />
    </React.StrictMode>
)