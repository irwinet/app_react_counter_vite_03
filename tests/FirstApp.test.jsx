/* eslint-disable no-undef */

import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp />', () => { 
    test('Debe de hacer match con el snapshot', () => {
        const title = "Titulo"
        const subTitle = 123
        const { container } = render(<FirstApp title={ title } subTitle= { subTitle } />);

        expect(container).toMatchSnapshot();
    });

    test("Debe de mostrar el titulo en un h1", () => {
        const title = "Titulo"
        const subTitle = 123
        const { container, getByText } = render(<FirstApp title={ title } subTitle= { subTitle } />);

        expect(getByText(title)).toBeTruthy();
        const h1 = container.querySelector('h1');
        // expect(h1.innerHTML).toBe(title);
        expect(h1.innerHTML).toContain(title);
    });
});