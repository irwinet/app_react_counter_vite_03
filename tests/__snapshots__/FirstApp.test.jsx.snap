// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Titulo
  </h1>
  <h1>
    Hola: Irwin
  </h1>
  <p>
    124
  </p>
  <p>
    Irwin
  </p>
</div>
`;
